### ISSUE TITLE
Replace `OrderedDict` with `dict` in Python 3.7+

### ISSUE THREAD

**Issue by StokesMIDE** [MideTechnology/ebmlite#97]
The library uses `collections.OrderedDict` when dumping EBML data in order to preserve element order. As of Python 3.7, standard dictionaries are explicitly insert-ordered (it was implementation-specific in 3.6). 

I propose using standard `dict` if running in Python 3.7 or later. Reasons:
* `OrderedDict` in Python 3.7+ does a substantial amount redundant work. Most of this is probably for strict backwards compatibility that `ebmlite` doesn't need.
* `str()` and `repr()` of `OrderedDict` are somewhat harder to read than standard `dict` (shows items as a list of (key,value) tuples, etc.).
* `pprint` doesn't format `OrderedDict` as cleanly (mostly for the same reasons as above).

**Comment by StokesMIDE** [comment]
The lowest impact way to change this is to simply add this to `core.py`:
```python
if sys.hexversion < 0x03070000:
    from collections import OrderedDict
else:
    OrderedDict = dict
```
I've made this change in branch [feature/97_dicts_in_py37](https://github.com/MideTechnology/ebmlite/tree/feature/97_dicts_in_py37) and it seems to have no problems, although it's a bit of a hack.

**Comment by StokesMIDE** [comment]
Implemented in #98
